src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfoOfT.cs

```diff
@@ -203,27 +201,25 @@ internal sealed class JsonPropertyInfo<T> : JsonPropertyInfo
         /// Create a <see cref="JsonPropertyInfo"/> for a given Type.
         /// See <seealso cref="JsonTypeInfo.PropertyInfoForTypeInfo"/>.
         /// </summary>
-        internal static JsonPropertyInfo CreateForSourceGenTypeInfo(
-            Type declaredPropertyType,
+        internal override void InitializeForTypeInfo(
+            Type declaredType,
             JsonTypeInfo runtimeTypeInfo,
             JsonConverter converter,
             JsonSerializerOptions options)
         {
-            JsonPropertyInfo<T> jsonPropertyInfo = new JsonPropertyInfo<T>();
-            jsonPropertyInfo.DeclaredPropertyType = declaredPropertyType;
-            jsonPropertyInfo.RuntimePropertyType = declaredPropertyType;
-            jsonPropertyInfo.ConverterStrategy = converter.ConverterStrategy;
-            jsonPropertyInfo.RuntimeTypeInfo = runtimeTypeInfo;
-            jsonPropertyInfo.ConverterBase = converter;
-            jsonPropertyInfo.Options = options;
-            jsonPropertyInfo.IsForTypeInfo = true;
-            jsonPropertyInfo.HasGetter = true;
-            jsonPropertyInfo.HasSetter = true;
+            DeclaredPropertyType = declaredType;
+            RuntimePropertyType = declaredType;
```

Comment:
Yes it will be needed for polymorphic support, although the value we set here will likely change. We already set this property today for non code-gen cases.